### PR TITLE
Add Card Layout with descriptions

### PR DESCRIPTION
**Pull request by jaronheard**
- Refactored CivicCardLayoutFull semantic HTML structure, split other components to new file, and use Emotion 10.
- Added a new layout (`CivicCardLayoutFullWithDescriptions`) which uses a newly added `cardMetaDescriptions` file to provide an annotated view showing each of the sections of a card, shown below

Screenshot_2019-08-17 Storybook

Especially interested in comments on the descriptions.

### CODE REVIEW

File: packages/component-library/src/CivicCard/CivicCardLayoutFull.js [hunk 1]
```diff
@@ -1,13 +1,17 @@
-import React, { Fragment } from "react";
+/** @jsx jsx */
+import { jsx, css } from "@emotion/core";
+import { Fragment } from "react";
```
Comment (jaronheard) [inline packages/component-library/src/CivicCard/CivicCardLayoutFull.js]: I wanted to use the `<> </>` Fragment syntax here, but in combination with the `/** @jsx jsx */` emotion syntax, I was having an issue.

If I recall if you `import React from "react"` it will work, but that line requires an `eslint ignore`

File: packages/component-library/src/CivicCard/cardMetaDescriptions.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      "REMOVE? REPLACED WITH METADATA QA - This section will point to the metadata API endpoint containing responses to the dataset metadata questionnaire"
+  },
+  metadataQA: {
+    title: "About This Data",
+    description:
+      "UNDER CONSTRUCTION - This section will point to the metadata API endpoint containing responses to the dataset metadata questionnaire"
+  },
+  resources: {
+    title: "Links and Resources",
+    description: "This section includes links and resources"
```
Comment (Dianna) [inline packages/component-library/src/CivicCard/cardMetaDescriptions.js]: I think this could be more descriptive. Links and resources to what?

File: packages/component-library/src/CivicCard/cardMetaDescriptions.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  },
+  analysis: {
+    title: "About This Analysis",
+    description:
+      "This section is a description of the analysis and includes a link to the data science notebooks and repository"
+  },
+  metadata: {
+    title: "About This Data",
+    description:
+      "REMOVE? REPLACED WITH METADATA QA - This section will point to the metadata API endpoint containing responses to the dataset metadata questionnaire"
```
Comment (Dianna) [inline packages/component-library/src/CivicCard/cardMetaDescriptions.js]: Yes? If it's not being used anymore. That could be a separate PR.

**Review by ghost — APPROVED** [review]
This looks good. The semantic html is definitely an improvement. Heads up, this PR does cause some critical and severe accessibility violations according to the Storybook checker. Just a couple of tiny improvements I'd recommend.

File: packages/component-library/src/CivicCard/cardMetaDescriptions.js (hunk 2 above)
Comment (jaronheard) [reply packages/component-library/src/CivicCard/cardMetaDescriptions.js]: Improved this section

File: packages/component-library/src/CivicCard/cardMetaDescriptions.js (hunk 3 above)
Comment (jaronheard) [reply packages/component-library/src/CivicCard/cardMetaDescriptions.js]: Will remove in a separate PR

File: packages/component-library/src/CivicCard/CivicCardLayoutFull.js (hunk 1 above)
Comment (jaronheard) [reply packages/component-library/src/CivicCard/CivicCardLayoutFull.js]: Ah, I dug into it.

You actually have to import React from "react" for the `<> </>`  syntax to work, because [@babel/plugin-transform-react-jsx](https://babeljs.io/docs/en/babel-plugin-transform-react-jsx) transforms it to React.fragment, so if you don't import React you will get an error.

File: packages/component-library/src/ChartTitle/ChartTitle.js [hunk 4]
```diff
@@ -29,10 +29,10 @@ const subtitleStyle = css`
...
 const ChartTitle = ({ title, subtitle }) =>
   title || subtitle ? (
-    <div>
-      {title ? <h3 className={titleStyle}>{title}</h3> : null}
-      {subtitle ? <span className={subtitleStyle}>{subtitle}</span> : null}
-    </div>
+    <figcaption>
+      {title ? <h2 className={titleStyle}>{title}</h2> : null}
+      {subtitle ? <h3 className={subtitleStyle}>{subtitle}</h3> : null}
+    </figcaption>
```
Comment (Dianna) [inline packages/component-library/src/ChartTitle/ChartTitle.js]: Check if this is proper usage of `figcaption` (which should be wrapped in a `figure`)

File: packages/component-library/src/CivicCard/CivicCardLayoutFull.js [hunk 5]
```diff
@@ -48,7 +49,7 @@ function CivicCardLayoutFull({ isLoading, data, cardMeta }) {
           <hr />
           <section id="tags">
             {cardMeta.tags.map((tag, index) => (
-              <Chip tag={tag} index={index} />
+              <Chip tag={tag} index={index} key={generate()} />
```
Comment (Dianna) [inline packages/component-library/src/CivicCard/CivicCardLayoutFull.js]: Nice

**Review by ghost — APPROVED** [review]
Looks great after that `figcaption` update 👍

File: packages/component-library/src/ChartTitle/ChartTitle.js (hunk 4 above)
Comment (jaronheard) [reply packages/component-library/src/ChartTitle/ChartTitle.js]: It wasn't quite right! Thankfully I had [@fstorr's awesome reference](https://fstorr.github.io/hack-oregon-civic-components/demo-card-visualization/) and I think I now have it sorted ... at least for this part.

Next step will be to see if I can capture the legend in the ChartContainer. I'll make an issue about that.
